src/controllers/connectionManager.ts

```diff
@@ -262,13 +262,16 @@ export default class ConnectionManager {
 	/**
 	 * Get the connection string for the provided connection Uri
 	 * @param connectionUri The connection Uri for the connection.
+	 * @param connectionDetails The connection info for the connection.
 	 * @param includePassword (optional) if password should be included in connection string.
 	 * @param includeApplicationName (optional) if application name should be included in connection string.
 	 * @returns connection string for the connection
 	 */
-	public async getConnectionString(connectionUri: string, includePassword: boolean = false, includeApplicationName: boolean = true): Promise<string> {
+	public async getConnectionString(connectionUri?: string, connectionDetails?: IConnectionInfo,
```

Comment:
You should have the first parameter be

`connectionContext: string | IConnectionInfo`

where you can pass it either a URI string or info object. Easier to use on the typescript side.